### PR TITLE
[YUNIKORN-1054] Add more queue metrics to Yunikorn

### PR DESCRIPTION
**Pull request by pbacsko**
### What is this PR for?
Adding more queue metrics to Yunikorn.

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-1054

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

**Comment by pbacsko** [comment]
Still in WIP. 
Discuss testing & `RMReleaseAllocationEvent.QueueNames`.

File: pkg/scheduler/objects/application_state.go [hunk 1]
```diff
@@ -143,49 +141,53 @@ func NewAppState() *fsm.FSM {
 				}
 			},
 			"leave_state": func(event *fsm.Event) {
-				event.Args[0].(*Application).clearStateTimer()
+				event.Args[0].(*Application).clearStateTimer() //nolint:errcheck
```
Comment (wilfred-s) [inline pkg/scheduler/objects/application_state.go:144]: Lets do this consistently:
* Remove all error checks on cast from all callback definitions including the string cast
* Add the nolint comment to all casts 
* Document at the start of the callback declaration the fact that arg 0 must be an app object and arg 1 a string. Doing something different will cause a run time panic.

**Review by wilfred-s — CHANGES_REQUESTED** [review]
Can we instead of passing the queue info all the way back up rely on the `partitionContext` code to trigger the queue metrics changes?
Every add and remove will go through the partition and in the partition code we have access to the application and thus the queue. We need to either increase or decrease the allocated resources in the queues on add and remove of the alloc. Check for the `removeNodeAllocation()` and `removeAllocation()` code in the partition context. Inside those we know the number of containers removed as we call `DecAllocatedResource()` with the sum of resources. We thus also know the count.
Similarly we can do the add of allocations.

I think that will simplify the change.

**Comment by pbacsko** [comment]
TODO:
* Pending apps per queue
* Killed apps per queue (needed?)

File: pkg/scheduler/partition.go [hunk 2]
```diff
@@ -1314,6 +1317,8 @@ func (pc *PartitionContext) removeAllocation(release *si.AllocationRelease) ([]*
 				zap.String("appID", appID),
 				zap.String("allocationId", uuid),
 				zap.Error(err))
+		} else {
+			metrics.GetQueueMetrics(queue.QueuePath).IncReleasedContainer()
```
Comment (wilfred-s) [inline pkg/scheduler/partition.go:1321]: I think this should be a GetQueuePath() call and not a direct access to the QueuePath

**Review by wilfred-s — CHANGES_REQUESTED** [review]
One minor nit: unlocked QueuePath access

**Review by wilfred-s — APPROVED** [review]
LGTM

**Pull request closed without merging**